src/tags/yaml-1.1/index.js

```diff
@@ -10,8 +10,18 @@ import { pairs } from './pairs.js'
 import { set } from './set.js'
 import { intTime, floatTime, timestamp } from './timestamp.js'
 
-const boolStringify = ({ value }) =>
-  value ? boolOptions.trueStr : boolOptions.falseStr
+const boolStringify = ({ value, sourceStr }) => {
+  const match = yaml11.filter(t => t.identify && t.identify(value))
+  const tagObj = match.find(t => !t.format)
+  if (sourceStr && !tagObj.test.test(sourceStr)) sourceStr = null
+  return value ? sourceStr ?? boolOptions.trueStr : sourceStr ?? boolOptions.falseStr
```

Comment:
If `sourceStr` is not set, this does a whole lot of useless extra work. Also, if there's a need to easily get at the right regexp, just extract them from the tag objects into top-level variables and use them from there.

Reply:
Good! done.